crates/pathfinder/build.rs

```diff
@@ -0,0 +1,23 @@
+//! Pathfinder build script.
+//!
+//! Just sets up `vergen` to query our git information for the build.
+
+fn main() {
+    // our Dockerfile is set up to a dependency only run, to cache a layer with all of the
+    // dependencies downloaded. during that "DEPENDENCY_LAYER" environment variable is set,
+    // and we shouldn't ask vergen to setup anything on that run to avoid this adding additional
+    // cache-miss reasons.
+
+    if std::env::var_os("DEPENDENCY_LAYER").is_some() {
+        return;
+    }
```

Comment:
If the commits after the second commit are dropped, and this is used with the dependency caching phase, it should not have any effect on the subsequent `cargo build --release` because this build.rs is touched.

Reply:
What's the intent here? Merging _all_ the commits, or only the first one?

If we want all of them I guess this check for DEPENDENCY_LAYER should also be removed...

Reply:
I think the jury is still out on if we can salvage the dep layer of dockerfile or not. All commits are required for a **working** docker build, but agree this should be dropped if it's not needed.

Reply:
On the intent, I remember trying to comment on it, but the intent was to keep the dep layer of dockerfile working. As in, if the deps stay the same, the old layers could be used on code-only modifications (after installing libgit2-dev). I was unable to find a working solution, so this is currently just a draft.

Reply:
This was useful in the end, resolving.